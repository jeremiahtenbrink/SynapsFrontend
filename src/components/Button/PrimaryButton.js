import React from "react";
import styled, { css } from "styled-components";
import PropTypes from "prop-types";
import BasicButton from "./Button.js";

/**
 * Button Component
 *
 * @component
 *
 * @example
 * return (
 *    <PrimaryButton  />
 *    )
 */
export const PrimaryButton = ( {
  text, height, width, type = "primary", size = "default", icon, shape, loading, block, color, appView, allFieldsValidated, ...props
} ) => {
  return ( <StyledAntdButton
    type={ type }
    size={ size }
    icon={ icon }
    shape={ shape }
    loading={ loading }
    block={ block && "block" }
    height={ height }
    width={ width }
    { ...props }
  >
    { text }
  </StyledAntdButton> );
};

PrimaryButton.propTypes = {
  text: PropTypes.string,
  type: PropTypes.oneOf( [
    "primary", "primaryCreateCard", "defaultCreateCard", "default", "dashed",
    "danger", "link", "darkgray",
  ] ),
  size: PropTypes.oneOf( [ "large", "default", "small" ] ),
  icon: PropTypes.string,
  shape: PropTypes.oneOf( [ "circle", "round" ] ),
  loading: PropTypes.bool,
  block: PropTypes.bool,
  background: PropTypes.string,
};

const StyledAntdButton = styled( BasicButton )`
  && {
    border: ${ props => {
  debugger;
  if( props.type === "defaultCreateCard" && props.appView ===
    "APP_VIEW_MOBILE" ){
    return "3px solid #4CB69F";
  }else if( props.type === "defaultCreateCard" && props.appView ===
    "APP_VIEW_DESKTOP" ){
    return "1px solid #343D58";
  }
} };

    height: ${ props => ( props.height ? props.height : "100%" ) };
    width: ${ props => ( props.width ? props.width : "100%" ) };

    color: ${ props => {
  debugger;
  if( props.appView === "APP_VIEW_MOBILE" ){
    if( props.type === "secondary" ){
      return "white";
    }else if( props.type === "defaultCreateCard" ){
      return "#4CB69F";
    }else if( props.type === "primaryCreateCard" ){
      return "#fff";
    }else{
      return "black";
    }
  }else{
    if( props.type === "defaultCreateCard" ){
      return "#343D58";
    }else if( props.type === "primaryCreateCard" ){
      return "#fff";
    }
  }
} };


    :active {
      background-color: ${ props => {
  debugger;
  if( props.type === "primaryCreateCard" && props.allFieldsValidated ){
    return "#235449";
  }else if( props.type === "defaultCreateCard" ){
    return "#6FEDB7";
  }
} };
      border: ${ props => {
  if( props.appView === "APP_VIEW_MOBILE" && props.allFieldsValidated ){
    if( props.type === "primaryCreateCard" ){
      return "3px solid #235449";
    }else if( props.type === "defaultCreateCard" ){
      return "3px solid #6FEDB7";
    }else{
      return 0;
    }
  }
} };
    }
    ${ props => {
  debugger;
  if( props.type === "primaryCreateCard" || props.type ===
    "defaultCreateCard" ){
    if( props.appView === "APP_VIEW_MOBILE" ){
      return "width: 136px; height: 42px; border-radius: 11px;";
    }else{
      return "width: 264px; height: 54px; border-radius: 33px;";
    }
  }
} }
    ${ props => {
  debugger;
  if( props.type === "primaryCreateCard" || props.type ===
    "defaultCreateCard" ){
    if( props.appView === "APP_VIEW_DESKTOP" ){
      return "font-size: 21px;";
    }
  }
} }
  }
`;

