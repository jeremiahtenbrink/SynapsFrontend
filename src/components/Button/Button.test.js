import React from 'react';
import renderer from 'react-test-renderer';

import {PrimaryButton} from './PrimaryButton.js';


test('StyledButton render correctly', () => {
    const tree = renderer.create(<PrimaryButton />).toJSON();
    expect(tree).toMatchSnapshot();
});

