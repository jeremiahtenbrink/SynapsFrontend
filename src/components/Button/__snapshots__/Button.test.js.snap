// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton render correctly 1`] = `
.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 4px solid white;
}

.c0.c0 p {
  font-family: "Source Sans Pro",serif;
  font-size: 28px;
  color: white;
  font-weight: 700;
  line-height: 1.1;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 70%;
}

.c0.c0 .one {
  top: 50%;
  left: 50%;
  opacity: .8;
  color: #f2f0ed;
}

.c0.c0 .two {
  top: 50.2%;
  left: 50.2%;
  opacity: .6;
  color: #edebe7;
}

.c0.c0 .three {
  top: 50.4%;
  left: 50.4%;
  opacity: .4;
  color: #e9e6e2;
}

.c0.c0 .four {
  top: 50.6%;
  left: 50.6%;
  opacity: .2;
  color: #e5e2dc;
}

.c0.c0 .five {
  top: 50.8%;
  left: 50.8%;
  opacity: .1;
  color: #e5e2dc;
}

<button
  className="ant-btn c0 ant-btn-round"
  onClick={[Function]}
  type="button"
/>
`;
